lib/evmone/eof.cpp

```diff
@@ -619,16 +619,16 @@ uint8_t get_eof_version(bytes_view container) noexcept
 
 EOFValidationError validate_eof(evmc_revision rev, bytes_view container) noexcept
 {
+    if (rev < EVMC_PRAGUE)
```

Comment:
I'm not very convinced about this change, and whether we should test validation for pre-Prague at all.

It made a bit more sense in 2022 EOF, when normal CREATEs were used to deploy EOF, but only after EOF activation. The check verified that CREATEs don't start to allow EOF before the fork.

Now CREATE/CREATE2 must always fail with EOF, so maybe these tests don't make sense anymore.

Reply:
It makes sense, and would help to avoid previous confusions in tests.